client/controller/shared.js

```diff
@@ -35,25 +37,86 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/**
+ * Loads a a css stylesheet into the page
+ * @param {string} cssUrl - a url to a css resource to be inserted into the page
+ * @param {Function} callback - a callback function to be called when the CSS has been loaded (after 500ms have passed).
+ */
+function loadCSS( cssUrl, callback = noop ) {
+	const link = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: cssUrl,
+	} );
+
+	const onload = () => {
+		if ( 'onload' in link ) {
+			link.onload = null;
+		}
+		callback( null, link );
+	};
+
+	if ( 'onload' in link ) {
+		link.onload = onload;
+	} else {
+		setTimeout( onload, 500 );
+	}
+
+	document.head.appendChild( link );
+}
+
 export function setSection( section ) {
 	return ( context, next = noop ) => {
 		context.store.dispatch( setSectionAction( section ) );
 
-		next();
+		loadSectionCSS( context, next );
 	};
 }
 
+export function loadSectionCSS( context, next ) {
+	const section = getSection( context.store.getState() );
+
+	if ( section.cssUrls && typeof document !== 'undefined' ) {
+		const cssUrl = context.isRTL ? section.cssUrls.rtl : section.cssUrls.ltr;
+
+		// TODO: handle adding styles in `state.documentHead.meta` instead (currently only supports setting all meta at once)
+		const currentLink = document.getElementById( 'section-css' );
+
+		// if the url has not changed, skip
+		if ( currentLink && currentLink.getAttribute( 'href' ) === cssUrl ) {
+			return next();
+		}
+
+		loadCSS( cssUrl, ( err, newLink ) => {
```

Comment:
Do we really need `err`? As far I can see the callback in the `loadCSS()` function always passes `null` for it.

Reply:
It's node's convention that callback always has first param as the error, in this case I don't have a preference either way/

Reply:
I wasn't aware of such convention, but still I don't think we should follow it blindly as it just adds confusion to our code.

Reply:
It's either that or a Promise. I wanted to avoid requiring using too many dependencies.

Reply:
I'm not sure to follow what you're saying @Tug. Could you clarify when you have a few?